app/models/pseud.rb

```diff
@@ -124,6 +124,22 @@ class Pseud < ApplicationRecord
     includes(:user)
   }
 
+  def self.check_pseud_coauthor?(pseud_id)
```

Comment:
in app/models/creatorship.rb:   


  validate :allowed_creator

  def allowed_creator
    return if Pseud.check_pseud_coauthor?(pseud_id)
    errors.add(:base,"Trying to add a invalid co creator")
  end

I only have the pseud_id and it feels heavy weight to find an object when I am only checking against an ID anyway.

Reply:
The first line of this function loads both the pseud and the user. Calling `Pseud.check_pseud_coauthor?(pseud_id)` instead of `pseud.check_pseud_coauthor?` actually requires more work by the database, since it forces both the pseud and user to be reloaded. (Which can be useful in some cases, but I doubt this isn't one of them.)

Reply:
I understand :)